src/test/java/com/toopher/HttpClientMock.java

```diff
@@ -65,22 +100,22 @@ public String getLastCalledData(String key) {
         } else {
             lastParams = req.getParams();
         }
-        HttpResponse resp = new BasicHttpResponse(new BasicStatusLine(HttpVersion.HTTP_1_1, expectedResponseStatus, null));
         BasicHttpEntity entity = new BasicHttpEntity();
+        if ( req == lastRequest && expectedUriResponses != null) {
+            expectedResponseBody = expectedUriResponses.get(req.getURI()).getResponseBody();
+            expectedResponseStatus = expectedUriResponses.get(req.getURI()).getStatusCode();
+        }
         try {
             entity.setContent(new ByteArrayInputStream(expectedResponseBody.getBytes("UTF-8")));
         } catch (UnsupportedEncodingException e) {
             e.printStackTrace();
         }
+        HttpResponse resp = new BasicHttpResponse(new BasicStatusLine(HttpVersion.HTTP_1_1, expectedResponseStatus, null));
         resp.setEntity(entity);
-
+        lastURI = req.getURI();
         T result;
-        try {
-            result = responseHandler.handleResponse(resp);
-        } catch (IOException e) {
-            result = null;
-            e.printStackTrace();
-        }
+        result = responseHandler.handleResponse(resp);
```

Comment:
I think you can combine the declaration and initalization: `T result = responseHandler.handleResponse(resp);`